From 44c03b18f15fc50bac6803aa1e65ce6a339234cf Mon Sep 17 00:00:00 2001
From: Saadi Maalem <saadi.maalem@intel.com>
Date: Fri, 20 Jun 2014 19:10:51 +0200
Subject: [PATCH 342/429] watchdog: change KWDG_MSI from 15 to 12

SCU sends MSI 15 when kernel watchdog expires.
Purpose is to trigger a BUG() on kernel side, to get
a call stack before resetting.

The MSI 15 is shared with i2c-6, which is used on EDISON.
Therefore any access to i2c-6 triggers a BUG()

To solve this problem, change the KWDG_MSI to 12.
This MSI is for i2c-3, which is unused on EDISON.

Signed-off-by: Saadi Maalem <saadi.maalem@intel.com>
---
 drivers/watchdog/intel_scu_watchdog_evo.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/watchdog/intel_scu_watchdog_evo.c b/drivers/watchdog/intel_scu_watchdog_evo.c
index 7b92247ed308..8aeecc4ead16 100644
--- a/drivers/watchdog/intel_scu_watchdog_evo.c
+++ b/drivers/watchdog/intel_scu_watchdog_evo.c
@@ -50,7 +50,7 @@
 #define STRING_COLD_RESET           "COLD_RESET"
 #define STRING_COLD_BOOT            "COLD_BOOT"
 
-#define EXT_TIMER0_MSI 15
+#define EXT_TIMER0_MSI 12
 
 #define IPC_WATCHDOG 0xF8
 
@@ -1228,7 +1228,7 @@ static int intel_scu_watchdog_init(void)
 		}
 	}
 
-	/* MSI #15 handler to dump registers */
+	/* MSI #12 handler to dump registers */
 	handle_mrfl_dev_ioapic(EXT_TIMER0_MSI);
 	ret = request_irq((unsigned int)EXT_TIMER0_MSI,
 		watchdog_warning_interrupt,
-- 
2.37.3

